### PR TITLE
Tab items/button

### PR DESCRIPTION
**Pull request by nabil-k**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [ ] android

### Description of changes

Made tabItems a button and passed along some props that button can use. tabItems can now use headerText to set the title of the tab and it can also use the icon prop to set the icon.

### Verification
Ran yarn build to check for any type errors and tested the changes in the tester app.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Tabs/src/TabsItem.tsx [hunk 1]
```diff
@@ -43,16 +41,6 @@ export const TabsItem = compose<TabsItemType>({
...
-    // Ensure focus is placed on button after click
-    const changeSelectionWithFocus = useOnPressWithFocus(componentRef, changeSelection);
-
-    /* RadioButton changes selection when focus is moved between each RadioButton and on a click */
-    const pressable = useAsPressable({
-      ...rest,
-      onPress: changeSelectionWithFocus,
-      onFocus: changeSelection,
-    });
-
```
Comment (dake3601) [inline packages/components/Tabs/src/TabsItem.tsx:55]: In develop, removing this breaks the button, does the button still work without this? (try with alerts)

File: packages/components/Tabs/src/Tabs.tsx [hunk 2]
```diff
@@ -95,7 +96,7 @@ export const Tabs = compose<TabsType>({
 
   settings,
   slots: {
-    root: View,
+    root: Stack,
```
Comment (dake3601) [inline packages/components/Tabs/src/Tabs.tsx:99]: What does the change from View to Stack do?

File: packages/components/Tabs/src/TabsItem.tsx (hunk 1 above)
Comment (nabil-k) [reply packages/components/Tabs/src/TabsItem.tsx:55]: When moving from pressable to button we can just use the buttons onClick property. I tested if the button clicking works when overriding the onClick property and it worked.

File: packages/components/Tabs/src/Tabs.tsx (hunk 2 above)
Comment (nabil-k) [reply packages/components/Tabs/src/Tabs.tsx:99]: I don't remember why I changed that lol. To be honest I'm not even sure what Stack does in general.

File: packages/components/Tabs/src/TabsItem.tsx (hunk 1 above)
Comment (dake3601) [reply packages/components/Tabs/src/TabsItem.tsx:55]: Is the button also clicked when you select it (without pressing)? As it is Tabs, the button must be selected with the arrows, does that still work?

File: packages/components/Tabs/src/Tabs.tsx (hunk 2 above)
Comment (dake3601) [reply packages/components/Tabs/src/Tabs.tsx:99]: We should probably check the difference before changing it.

File: packages/components/Tabs/src/TabsItem.tsx (hunk 1 above)
Comment (nabil-k) [reply packages/components/Tabs/src/TabsItem.tsx:55]: Made a new commit that shows onFocus being overloaded. We can use that prop as an entry to handle selecting the tab onFocus :)

**Pull request closed without merging**